opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -760,7 +761,8 @@ def start_as_current_span(
             record_exception=record_exception,
             set_status_on_exception=set_status_on_exception,
         )
-        return self.use_span(span, end_on_exit=True)
+        with self.use_span(span, end_on_exit=True) as span_context:
+            yield span_context
 
     def start_span(  # pylint: disable=too-many-locals
```

Comment:
Does this same decorator pattern work with `start_span()`? I would expect them to behave the same

Reply:
It doesn't use `@contextmanager`, so no. And it should not